lib/api/src/sys/instance.rs

```diff
@@ -166,6 +175,14 @@ impl Instance {
             })
             .collect::<Exports>();
 
+        // If the memory is imported then also export it for backwards compatibility reasons
+        // (many will assume the memory is always exported) - later we can remove this
+        if exports.get_memory("memory").is_err() {
```

Comment:
I don't understand how this is required for backward compatibility while it's new code

Reply:
We should not have the logic on the main API

Reply:
discussed, stays for now